### PR TITLE
Fixes related to `address` inside conf.

### LINKED ISSUE #231: @Harshit933 also in this case it is wrong, we should set it to `None` and avoid to bind the server on a specific peer.
@Harshit933 also in this case it is wrong, we should set it to `None` and avoid to bind the server on a specific peer.

However it is unrelated to this PR

_Originally posted by @vincenzopalazzo in https://github.com/vincenzopalazzo/lampo.rs/pull/229#discussion_r1614519284_

### PR DESCRIPTION
**Pull request by Harshit933**
There are two things that are addressed here : 

1. Throw a warning that the node is not reachable if the `announce-addr` inside `conf` is not present.
2. Stop the node in case the `announce-addr` mentioned inside the `conf` is wrong or could not be binded.
Fixes #231

### CODE REVIEW

File: lampod-cli/src/main.rs [hunk 1]
```diff
@@ -155,7 +155,9 @@ fn run(args: LampoCliArgs) -> error::Result<()> {
         std::process::exit(0);
     })?;
 
-    let workder = lampod.listen().unwrap();
+    let workder = lampod.listen().map_err(|err| {
+        return error::anyhow!("{err}");
+    })?;
```
Comment (vincenzopalazzo) [inline lampod-cli/src/main.rs:160]:
```suggestion
    let workder = lampod.listen()?;
```

Please, do not apply monkey change

File: lampod/src/ln/peer_manager.rs [hunk 2]
```diff
@@ -132,12 +132,13 @@ impl LampoPeerManager {
...
-        let addr = self
-            .conf
-            .announce_addr
-            .clone()
-            .unwrap_or_else(|| "127.0.0.1".to_string());
-        std::thread::spawn(move || {
+        let addr = if let Some(addr) = self.conf.announce_addr.clone() {
+            addr
+        } else {
+            log::warn!("Address not specified inside conf, node is not reachable");
```
Comment (vincenzopalazzo) [inline lampod/src/ln/peer_manager.rs:138]:
```suggestion
```

Why a warning? if I want to stay private I do  not announce the address

File: lampod/src/ln/peer_manager.rs [hunk 3]
```diff
@@ -132,12 +132,13 @@ impl LampoPeerManager {
...
+            return Ok(());
+        };
```
Comment (vincenzopalazzo) [inline lampod/src/ln/peer_manager.rs:140]:
```suggestion
        let Some(addr) = self.conf.announce_addr else {
            log::warn!("Address not specified inside conf, node is not reachable");
            return Ok(());
        };
```

Better usage of rust

File: lampod/src/ln/peer_manager.rs [hunk 4]
```diff
@@ -203,6 +204,7 @@ impl LampoPeerManager {
             }
             result
         });
+        result.join().unwrap()?;
```
Comment (vincenzopalazzo) [inline lampod/src/ln/peer_manager.rs:207]: Why this change? you are blocking lampo so from this join the code will never return, Have you tested the code?

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]

File: lampod/src/ln/peer_manager.rs (hunk 2 above)
Comment (Harshit933) [reply lampod/src/ln/peer_manager.rs:138]: Should I put a `log::info!` or completely removed this line?

Comment (vincenzopalazzo) [reply lampod/src/ln/peer_manager.rs:138]: There is no needed to add this message imho

File: lampod-cli/src/main.rs (hunk 1 above)
Comment (Harshit933) [reply lampod-cli/src/main.rs:160]: Thanks, changed

File: lampod/src/ln/peer_manager.rs (hunk 4 above)
Comment (Harshit933) [reply lampod/src/ln/peer_manager.rs:204]: I still need to see this, Will do in this week!

Comment (vincenzopalazzo) [reply lampod/src/ln/peer_manager.rs:204]: Take your time, this will be in the august release

**Comment by vincenzopalazzo** [comment]
Closing because inactive
